doc/cabal-package.rst

```diff
@@ -801,18 +800,45 @@ Library
 
     Build information for libraries.
 
-    A package can have one default library, which is identified by omitting the
-    ``name`` argument.
+    A package can include zero or more library components, being an optional
+    'main' unnamed library (which is identified by having no ``name`` argument)
+    and/or zero or more 'subsidiary' or 'ancilliary' named libraries (referred
+    to as 'sublibraries', which are identified by their ``name`` argument). A
+    sublibrary cannot have the same name as its package.
 
-    Starting with Cabal 2.0, sub-library components can be defined by setting
-    the ``name`` field to a name different from the current package's name; see
-    section on :ref:`Sublibraries <sublibs>` for more information. By
-    default, these sub-libraries are private and internal. Since Cabal 3.0,
-    these sub-libraries can also be exposed and used by other packages. See the
-    :pkg-field:`library:visibility` field and :ref:`Multiple Public Libraries
-    <publicsublibs>` for more information.
+    A sublibrary has a 'visibility' which can be 'public' or 'private' (the
+    default) - see the :pkg-field:`library:visibility` field. The 'main'
+    unnamed library is always public. A private sublibrary may be referred to as
+    an 'internal' library, because other packages cannot depend upon it.
 
-The library section should contain the following fields:
+    Before Cabal 3.0, all sublibraries were internal libraries. Before Cabal 2.0
+    a package could not include sublibraries.
+
+    A package being able to include more than one public library separates the
+    unit of distribution (the package) from the unit of buildable code (the
+    library). This is useful for Haskell projects with many libraries that are
+    distributed together as it avoids duplication and potential inconsistencies.
+
+    See :ref:`Sublibraries - Examples <sublibs>` for examples.
+
+A library section should contain the following fields:
+
+.. pkg-field:: visibility: visibility specifiers
+
+    :since: 3.0
+
+    :default:
+        ``private`` for sublibraries. Cannot be set for the 'main' unnamed
+        library, which is always public.
```

Comment:
```suggestion
        ``private`` for sublibraries. Cannot be set for the main
        library, which is always public.
```
As above, I don't think "main" needs to be quoted here and unnamed doesn't add any extra detail.

Reply:
As above, this was my attenpt to avoid chosing sides in the named/unamed v main/sub debate.

Reply:
I have now taken a different approach to this, by changing the introduction of terms.